http/handler_tx.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/go-chi/chi/v5"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+type txHandler struct{}
+
+type CreateTxResponse struct {
+	ID uint64 `json:"id"`
+}
+
+func (h *txHandler) NewTxn(rw http.ResponseWriter, req *http.Request) {
+	db := req.Context().Value(dbContextKey).(client.DB)
+	txs := req.Context().Value(txsContextKey).(*sync.Map)
+	readOnly, _ := strconv.ParseBool(req.URL.Query().Get("read_only"))
+
+	tx, err := db.NewTxn(req.Context(), readOnly)
+	if err != nil {
+		responseJSON(rw, http.StatusBadRequest, errorResponse{err})
+		return
+	}
+	txs.Store(tx.ID(), tx)
```

Comment:
question: I can't spot anything that 'expires' long-lived transactions which have likely been forgotten by the client.  Is that true?

I think some datastore implementations do this (not fully sure thought), meaning that the server will think a txn is alive, when the datastore has decided it is dead - are we relying on this to minimise expenses and handle errors?

question: If the above is true, are we sure we are happy tolerating a forever growing map of txns in the short-medium term?

Reply:
This is a great point. I think a good long term solution would be having a shared transaction pool within the database that allows the user to specify limits and timeouts.

This will also enable us to fix the "leaky" `datastore.Txn` interface. Exposing the `MultiStore` interface on transactions is currently a pain point for the http client.

In the short-term I think this will be okay. I think only advanced users will interact with the transaction system, and we can make a note of the current short-comings somewhere.

Reply:
Sounds sensible, sorry, I could of asked this earlier but didnt - could you please open a ticket to sort this out?

Reply:
Created issue #1843

Reply:
Cheers :)